src/data/actions/loginForm.js

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+import qs from 'query-string';
+
+import {
+  LOGIN_FAILURE,
+  LOGIN_REQUEST,
+  LOGIN_SUCCESS,
+} from '../constants/loginForm';
+
+const setLoginPending = loading => ({
+  type: LOGIN_REQUEST,
+  payload: { loading },
+});
+
+const setLoginSuccess = isLoginSuccess => ({
+  type: LOGIN_SUCCESS,
+  payload: { isLoginSuccess },
+});
+
+const seterror = error => ({
+  type: LOGIN_FAILURE,
+  payload: { error },
+});
+
+const authenticate = (email, passphrase, callback) => {
+  const dataToSend = {
+    grant_type: 'password',
+    username: email,
+    password: passphrase,
+    client_id: 'c8OHQhvyy1paUgrmVyNX18lC2AkgXPQdIF5xE7EY',
+    token_type: 'jwt',
+  };
+
+  const authUrl = 'http://localhost:18000/oauth2/access_token';
+  axios.post(authUrl, qs.stringify(dataToSend))
+    .then((response) => {
+      localStorage.setItem('access_token', response.data.access_token);
+      return callback('');
+    })
+    .catch(error => (
+      callback(error)
+    ));
+};
+
+const login = (email, password) => (
+  (dispatch) => {
+    dispatch(setLoginPending(true));
+    // follow authentication process for this
+    authenticate(email, password, (error) => {
+      dispatch(setLoginPending(false));
+      if (!error) {
+        dispatch(setLoginSuccess(true));
+        dispatch(seterror(''));
+      } else {
+        dispatch(setLoginSuccess(false));
+        dispatch(seterror(error.message));
+      }
```

Comment:
This logic can be cleaned up a bit once you refactor to use the API service layer. For example, you should only dispatch `setLoginSuccess` if the request was successful. Here, you're dispatching even if there was an error. Likewise for dispatching the `seterror` function. Refer to the actions/courseEnrollments.js file for an example.

Reply:
My reasoning behind this was to make debugging more verbose. If there was no error I believe we should dispatch that it was successful and dispatch a `null` error. If a user fails to login the first time, and successfully the second time then the error from the first attempt will remain. I agree that we should only dispatch only `seterror` if there was an error found though.

Reply:
@jonalexvasquez That logic can be handled by the reducer without needing to dispatch `LOGIN_SUCCESS` and `LOGIN_REQUEST` actions that take parameters. On `LOGIN_REQUEST`, you can reset `error` to `null`. See the [course enrollments reducer](https://github.com/edx/edx-portal/blob/master/src/data/reducers/courseEnrollments.js#L15) for an example.